src/DisputeHandler.ts

```diff
@@ -55,164 +72,281 @@ class DisputeHandler {
+            )!;
+        return {
+            genesisStateSnapshot: outputStateSnapshot,
+            latestStateSnapshot: this.agreementManager.getSnapShot(
+                forkCnt,
+                transactionCnt
+            )!,
+            outputStateSnapshot: outputStateSnapshot,
+            milestoneSnapshots: [],
+            latestStateStateMachineState: latestStateStateMachineState,
+            joinChannelBlocks: this.agreementManager.getJoinChannelChain(),
+            timestamp: 0,
+            previousDispute: prevDispute,
+            previousDisputeTimestamp: prevDisputeTimestamp
+        };
     }
 
-    public async disputeBlockTooFarInFuture(
-        BlockSigned: SignedBlockStruct
-    ): Promise<void> {
-        const proof = ProofManager.createBlockTooFarInFutureProof(BlockSigned);
-        const block = EvmUtils.decodeBlock(BlockSigned.encodedBlock);
-        return this.createDispute(
-            block.transaction.header.forkCnt,
-            NO_PARTICIPANT_TO_FOLD,
-            INITIAL_TRANSACTION_COUNT,
-            [proof]
-        );
+    private collectStateProof(forkCnt: number): StateProofStruct {
+        const forkProof =
+            this.agreementManager.forkService.getForkProof(forkCnt)!;
+        const signedBlocks =
+            this.agreementManager.getForkProofSignedBlocks(forkCnt);
+        const stateProof: StateProofStruct = {
+            forkProof: forkProof,
+            signedBlocks: signedBlocks
+        };
+        return stateProof;
     }
 
-    public onDispute(dispute: DisputeStruct): Promise<void> {
-        this.setForkDisputed(Number(dispute.forkCnt));
-        return this.rechallengeRecursive(dispute);
+    private createDefaultDisputeStruct(): {
+        dispute: DisputeStruct;
+        timeout: TimeoutStruct;
+    } {
+        const timeout: TimeoutStruct = {
+            participant: ethers.ZeroAddress,
+            blockHeight: ethers.MaxUint256,
+            minTimeStamp: ethers.MaxUint256,
+            forkCnt: 0,
+            isForced: false,
+            previousBlockProducer: ethers.ZeroAddress,
+            previousBlockProducerPostedCalldata: false
+        };
+
+        let dispute: DisputeStruct = {
+            channelId: this.channelId,
+            genesisStateSnapshotHash: "",
+            latestStateSnapshotHash: "",
+            stateProof: {
+                forkProof: {
+                    forkMilestoneProofs: []
+                },
+                signedBlocks: []
+            },
+            fraudProofs: [],
+            onChainSlashes: [],
+            onChainLatestJoinChannelBlockHash: "",
+            outputStateSnapshotHash: "",
+            exitChannelBlocks: [],
+            disputeAuditingDataHash: "",
+            disputer: this.signerAddress,
+            disputeIndex: 0,
+            previousRecursiveDisputeIndex: ethers.MaxUint256,
+            timeout,
+            selfRemoval: false
+        };
+        return { dispute, timeout };
```

Comment:
Not sure why such a function is needed. These things may make sense in tests, but here I don't see any rational logic for these default structures that can not exist and are invalid - submitting this on-chain would just get you slashed